src/renderer/shared/ui-kit/Select/Select.tsx

```diff
@@ -44,161 +91,211 @@ const Root = <T extends string>({
-          <Surface
-            elevation={1}
+    <Context.Provider value={contextValue}>
+      <ComboboxProvider
+        open={isOpen}
+        value={onSearch ? searchValue : ''}
+        setValue={value => {
+          if (onSearch) {
+            setSearchValue(value);
+            onSearch(value);
+          }
+        }}
+      >
+        {!isOpen ? (
+          <ComboboxDisclosure clickOnEnter={true} clickOnSpace={true} className="w-full rounded-md">
+            <button
+              ref={triggerRef}
+              className={cnTw(
+                'box-border cursor-pointer border px-2.75 text-text-secondary',
+                'w-full rounded-md text-left text-footnote hover:shadow-card-shadow',
+                {
+                  'h-8.5': height === 'sm',
+                  'h-10.5': height === 'md',
+                  'border-filter-border bg-input-background text-text-primary': theme === 'light',
+                  'border-border-dark bg-background-dark text-white': theme === 'dark',
+                  'cursor-not-allowed bg-input-background-disabled text-text-tertiary': disabled,
+                  'border-filter-border-negative': invalid,
+                },
+              )}
+              disabled={disabled}
+              onClick={() => {
+                !disabled && onOpenChange(true);
+              }}
+            >
+              {valueNode || selectedItemContent || <span className="text-text-secondary">{placeholder}</span>}
+            </button>
+          </ComboboxDisclosure>
+        ) : (
+          <Combobox
+            autoFocus
+            placeholder={placeholder}
+            readOnly={!onSearch}
             className={cnTw(
-              'z-50 flex flex-col',
-              'h-max max-h-(--radix-popper-available-height) min-w-20',
-              'origin-(--radix-popper-transform-origin) overflow-hidden duration-100 animate-in fade-in zoom-in-95',
-              {
-                'border-border-dark bg-background-dark': theme === 'dark',
-              },
+              'min-h-[34px] w-full rounded-md border-none px-3 outline-1 placeholder:text-footnote placeholder:text-text-secondary focus-visible:outline-2',
+              { 'h-10.5': height === 'md' },
+              { 'cursor-default': !onSearch },
             )}
-          >
-            <ScrollArea>
-              <RadixSelect.Viewport asChild>
-                <div className="flex flex-col gap-y-1 p-1">{children}</div>
-              </RadixSelect.Viewport>
-            </ScrollArea>
-          </Surface>
-        </RadixSelect.Content>
-      </RadixSelect.Portal>
-    </ThemeProvider>
+            onBlur={() => {
+              onOpenChange(false);
+            }}
+          />
+        )}
+        <ComboboxPopover
+          gutter={8}
+          sameWidth
+          className={cnTw(
+            'relative z-50 flex max-h-[min(var(--popover-available-height,300px),300px)] flex-col overflow-auto overscroll-contain rounded-md border p-1',
```

Comment:
```suggestion
            'relative z-50 flex max-h-[300px] flex-col overflow-auto overscroll-contain rounded-md border p-1',
```

better to have a fixed value but it conflicts with dialogs when overflows